privacy-on-beam/pbeam/aggregations.go

```diff
@@ -450,6 +483,156 @@ func clampNegativePartitionsFloat64Fn(v beam.V, r float64) (beam.V, float64) {
 func convertFloat32ToFloat64Fn(z beam.Z, f float32) (beam.Z, float64) {
 	return z, float64(f)
 }
+
 func convertFloat64ToFloat64Fn(z beam.Z, f float64) (beam.Z, float64) {
 	return z, f
 }
+
+// Turn PCollection<V> into PCollection <V,0>.
+func newPrepareAddPartitionsFn(vKind reflect.Kind) interface{} {
+	var fn interface{}
+	switch vKind {
+	case reflect.Int64:
+		fn = prepareAddPartitionsInt64Fn
+	case reflect.Float64:
+		fn = prepareAddPartitionsFloat64Fn
+	default:
+		log.Exitf("pbeam.newPrepareAddPartitionsFn: vKind(%v) should be int64 or float64", vKind)
+	}
+	return fn
+}
+
+// Turn PCollection<V> into either PCollection <V,[]int64{0}> or PCollection <V,[]float64{0}>.
+func newPrepareAddMeanPartitionsFn(vKind reflect.Kind) interface{} {
+	var fn interface{}
+	switch vKind {
+	case reflect.Int64:
+		fn = prepareAddPartitionsMeanInt64Fn
+	case reflect.Float64:
+		fn = prepareAddPartitionsMeanFloat64Fn
+	default:
+		log.Exitf("pbeam.newPrepareAddPartitionsFn: vKind(%v) should be int64 or float64", vKind)
+	}
+	return fn
+}
+
+func prepareAddPartitionsInt64Fn(partitionKey beam.X) (k beam.X, v int64) {
+	return partitionKey, 0
+}
+
+func prepareAddPartitionsFloat64Fn(partitionKey beam.X) (k beam.X, v float64) {
+	return partitionKey, 0
+}
+
+func prepareAddPartitionsMeanInt64Fn(partitionKey beam.X) (k beam.X, v []int64) {
+	return partitionKey, []int64{}
+}
+
+func prepareAddPartitionsMeanFloat64Fn(partitionKey beam.X) (k beam.X, v []float64) {
+	return partitionKey, []float64{}
+}
+
+// Function for sum and mean. Drop unspecified partitions.
+func dropUnspecifiedPartitions(s beam.Scope, partitionsCol beam.PCollection, pcol PrivatePCollection, partitionEncodedType beam.EncodedType) beam.PCollection {
+	partitionMap := beam.Combine(s, newPartitionsHashMapFn(partitionEncodedType), partitionsCol)
+	return beam.ParDo(s, prunePartitionsFn, pcol.col, beam.SideInput{Input: partitionMap})
+
+}
+
+// Function for count and distinct_id. Drop unspecified partitions.
+func dropUnspecifiedPartitionsForCount(s beam.Scope, partitionsCol beam.PCollection, pcol PrivatePCollection, partitionFullType typex.FullType) beam.PCollection {
+	partitionEncodedType := beam.EncodedType{partitionFullType.Type()}
+	partitionMap := beam.Combine(s, newPartitionsHashMapFn(partitionEncodedType), partitionsCol)
+	return beam.ParDo(s, newPrunePartitionsFnForCount(partitionFullType), pcol.col, beam.SideInput{Input: partitionMap})
+}
+
+type mapAccum struct {
+	// Key is string representation of partition encoding.
+	// Value is true.
+	PartitionMap PMap
+}
+
+// Make hashmaps for partitions.
+type PartitionsHashMapFn struct {
```

Comment:
nit: Maybe just "Map" instead of "HashMap" since that's the naming used in Go.

(Also in other places)